api/openapi.yaml

```diff
@@ -0,0 +1,1026 @@
+openapi: 3.1.0
```

Comment:
Why do we need this file ?

Reply:
I would like to use a library like https://github.com/cdimascio/express-openapi-validator to validate the requests/responses based on the openapi spec (the file comes from https://github.com/ollama/ollama/pull/5040)

Reply:
I would suggest another approach. We should write our own `ai-lab-spec.yaml` file with the `open api` specification and use open source tool like [openapi-generator.tech](https://openapi-generator.tech/) or [express-openapi](https://www.npmjs.com/package/express-openapi) to directly generate the server.

## Why ?

This will ensure type safety of our implementation matching the open api spec. Thus allowing users (E.g. quarkus) to generate a client using tools, to directly be able to uses our API.

## Personal opinion

This is an important architectural choice, and should be discussed before using/choosing any framework.

Reply:
As discussed with @jeffmaury and @slemeur , we would like to implement the Ollama api, thus the use of this specific openapi spec and not a ai-lab specific one.

I already looked at https://openapi-generator.tech/, but there is only one node server generator, in beta, which writes Javascript (not Typescript) code, which seems not optimal. I will have a look at https://www.npmjs.com/package/express-openapi.

I completely agree with your constraints in `Why?`, which would also be respected with the https://github.com/cdimascio/express-openapi-validator tool and the openapi spec provided by ollama. The only difference is that the creation of the server and the definition of the routes would be to be written by ourselves, and the types would be generated by https://github.com/openapi-ts/openapi-typescript. Type safety would be ensured by https://github.com/cdimascio/express-openapi-validator in real-time. In any case, the clients can use the ollama openapi spec to generate their client code. And because it is ollama compatible, the client is most probably already generated somewhere.